scripts/change_alias.py

```diff
@@ -10,57 +41,117 @@ class ChangeAliases:
-        Initialize object with prefix name and elasticsearch instance
-        :param name: prefix name
-        :param es_staging: elasticsearch instance
+        Initialize the tool
+        :param es_hosts: Elastic Search host names as a string separated by ";"
+        :param es_index_prefix: the index prefix which indicates the build of indices
         """
-        self.name = name
-        self.es_staging = es_staging
+        hosts = es_hosts.split(";")
+        logger.info("Command line parameters")
+        logger.info("Hosts: " + str(hosts))
+        if es_index_prefix:
+            logger.info("Index_prefix:" + es_index_prefix)
+        self.hosts = hosts
+        self.es_index_prefix = remove_underscore_from_end_prefix(es_index_prefix)
+        self.es = Elasticsearch(hosts)
+        self.current_aliases = dict()
 
     def run(self):
         """
         Main function to run aliases change
         """
-        print("Current aliases in use: ")
-        old_index_prefix = print_current_aliases(self.es_staging)
-        # Exit if user try to change aliases to indexes with the same prefix
-        if old_index_prefix == self.name:
-            print("Prefix is already in use, exiting!")
-            sys.exit(0)
+        self.get_current_aliases()
+        # no prefix parameter given, display current aliases
+        if not self.es_index_prefix:
+            logger.info("Current aliases in use: ")
+            duplicate = set(INDICES)
+            for alias, index in self.current_aliases.items():
+                logger.info("{} -> {}".format(alias, index))
+                duplicate.remove(alias)
+            if duplicate:
+                logger.warning(f"Some alias {str(duplicate)} has no corresponding index")
         else:
-            for index in INDICES:
-                new_index = "{}{}".format(self.name, index)
-                old_index = "{}{}".format(old_index_prefix, index)
-                self.change_aliases(new_index, old_index, index)
-            print("New aliases: ")
-            print_current_aliases(self.es_staging)
+            # generate the indices list according to the given index prefix
+            index_to_match = dict()
+            for alias in INDICES:
+                index_to_match[f"{self.es_index_prefix}_{alias}"] = alias
+            # get all indices from the given host
+            all_indices = self.get_all_indices()
+            # the action list which contains all actions required to make the desired changes
+            actions = list()
+            # iterate all indices on the given ES server
+            for index in all_indices:
+                # current index needs to be assigned to one of aliases
+                if index in index_to_match:
+                    alias = index_to_match[index]
+                    # check whether the alias has been already assigned
+                    if alias in self.current_aliases:
+                        current_alias = self.current_aliases[alias]
```

Comment:
Variables naming here is really confusing, could you rename current_alias on something more meaningful. As I understand current_alias variable will have name on index alias points to

Reply:
Also there are too many nested conditions and it's really hard to read. You might improve it like this:
1. alias = index_to_match.get(index, None)
2. if alias in self.current_aliases

Reply:
get the idea.
However the statement "if index in index_to_match:" can not be removed as it is the main condition to add new alias

Reply:
Agree.